resolver/src/p2p/fetcher.rs

```diff
@@ -270,7 +270,7 @@ impl<E: Clock + GClock + Rng + Metrics, P: PublicKey, Key: Span, NetS: Sender<Pu
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "mocks"))]
```

Comment:
I don't think this is required? `mocks` is now automatically enabled for tests (and this would allow someone to bypass tests unknowingly)?

Maybe I'm missing something...